src/main/java/org/sasanlabs/service/vulnerability/fileupload/UnrestrictedFileUpload.java

```diff
@@ -10,31 +18,149 @@
 import org.sasanlabs.service.vulnerability.bean.GenericVulnerabilityResponseBean;
 import org.sasanlabs.vulnerability.types.VulnerabilitySubType;
 import org.sasanlabs.vulnerability.types.VulnerabilityType;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
 
-/** @author KSASAN preetkaran20@gmail.com */
+/**
+ * @author KSASAN preetkaran20@gmail.com Special Thanks
+ *     <p>https://bezkoder.com/spring-boot-file-upload/
+ *     <p>https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects
+ *     <p>https://www.youtube.com/watch?v=CmF9sEyKZNo
+ */
 @VulnerableAppRestController(
         descriptionLabel = "UNRESTRICTED_FILE_UPLOAD_VULNERABILITY",
         value = "UnrestrictedFileUpload",
         type = {VulnerabilityType.UNRESTRICTED_FILE_UPLOAD})
 public class UnrestrictedFileUpload {
+    private final Path root;
 
-    /** https://bezkoder.com/spring-boot-file-upload/ READ it */
+    private static final String STATIC_FILE_LOCATION = "upload";
+    private static final String BASE_PATH = "static";
+    private static final Random RANDOM = new Random(new Date().getTime());
+
+    public UnrestrictedFileUpload() throws IOException, URISyntaxException {
+        URI uploadDirectoryURI =
+                new URI(
+                        Thread.currentThread()
+                                        .getContextClassLoader()
+                                        .getResource(BASE_PATH)
+                                        .toURI()
+                                + FrameworkConstants.SLASH
+                                + STATIC_FILE_LOCATION);
+        root = Paths.get(uploadDirectoryURI);
+        if (!root.toFile().exists()) {
```

Comment:
We could probably think of using something like
IOUtils/FileUtils to handle file operations , off-load some of the headaches

Reply:
Actually i think Directory handling's best way is to use NIO Api's, FileUtils can help with file operation. Other part of URI handling, i am not sure what is the alternative to get the path to resources folder easily.

Reply:
Yeah you're right , this should be fine :+1: